docs/architecture/sdk/server-bindings.md

```diff
@@ -0,0 +1,59 @@
+# Server Bindings
+
+The SDK currently uses auto-generated bindings for the server. The bindings are generated using the
+[`openapi-generator`][openapi] to generate the Rust bindings from the server OpenAPI specifications.
+These bindings are regularly updated to ensure they stay in sync with the server.
+
+The bindings are exposed as multiple crates, one for each backend service:
+
+- `bitwarden-api-api`: For the `Api` service, which contains most of the server side functionality.
+- `bitwarden-api-identity`: For the `Identity` service, which is used for authentication.
+
+When performing any API calls the goal is to use the generated bindings as much as possible. This
+ensures any changes to the server are accurately reflected in the SDK. The generated bindings are
+stateless, and always expects to be provided a `Configuration` instance. The SDK exposes these under
+the `get_api_configurations` function on the `Client` struct. `get_api_configurations` also
+refreshes the authentication token if required.
+
+```rust
+// Example API call
+let config: &ApiConfigurations = client.get_api_configurations().await;
+let response: SyncResponseModel =
+    bitwarden_api_api::apis::sync_api::sync_get(&config.api, exclude_subdomains).await?;
+```
+
+It's highly recommended to not expose the request and response models of the generated bindings and
```

Comment:
❓ Do we want to suggest something different if we intend to manage memory as much as possible? Should we mention memory security at all?